### ISSUE TITLE
Get error when trying to log in 

### ISSUE THREAD

**Issue by paulvancity** [boxing/boxrec#252]
`const boxrec = require("boxrec").Boxrec;

const cookieJar = await boxrec.login('myuser', 'mypass');`

Results in this error:

SyntaxError: await is only valid in async function
    at wrapSafe (internal/modules/cjs/loader.js:979:16)
    at Module._compile (internal/modules/cjs/loader.js:1027:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47

**Comment by mikedidomizio** [comment]
You have to use `await` within an `async` function

https://stackoverflow.com/questions/49432579/await-is-only-valid-in-async-function

Something like this should work

```
(async() => {
  const boxrec = require("boxrec").Boxrec;
  const cookieJar = await boxrec.login('myuser', 'mypass');
})();
```

**Comment by paulvancity** [comment]
Thank you, that seemed to work but when I try a query I get an error:


`(async() => {
    const boxrec = require("boxrec").Boxrec;
    const cookieJar = await boxrec.login('myuser', 'mypass');

    gennadyGolovkin = await boxrec.getPersonById(cookieJar, 356831);

    console.log(gennadyGolovkin.name); // Gennady Golovkin

  })();`


`(node:185649) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'filter' of undefined
    at /var/www/node/node_modules/cheerio/lib/api/traversing.js:30:42
    at Array.reduce (<anonymous>)
    at module.exports.exports.find (/var/www/node/node_modules/cheerio/lib/api/traversing.js:29:30)
    at Function.<anonymous> (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:741:66)
    at step (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:32:23)
    at Object.next (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:13:53)
    at fulfilled (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:4:58)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:185649) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:185649) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
`

**Comment by mikedidomizio** [comment]
I'm addressing this but it's fixing a bunch of things.  No ETA,  I want to say a few days but it requires me being free of everything.
This also won't fix that BoxRec has changed a bunch since.  I'm sure there's a lot of updating that's required.

**Comment by paulvancity** [comment]
> I'm addressing this but it's fixing a bunch of things.  No ETA,  I want to say a few days but it requires me being free of everything.
> 
> This also won't fix that BoxRec has changed a bunch since.  I'm sure there's a lot of updating that's required.

Thank you so much Mike 🙏

**Comment by mikedidomizio** [comment]
Try version 7.1.4 of this package.  It should fix requests.

`npm install --save boxrec@7.1.4`

or

`yarn add boxrec@7.1.4`

**Comment by paulvancity** [comment]
Hi Mike, 

thank you so much for the update but when I try now it just hangs at login.

Using node version v14.15.0

Thank you!

**Comment by paulvancity** [comment]
I still seem to be getting the error

├─┬ boxrec@7.1.4
│ ├─┬ boxrec-requests@5.1.1

Node Version: v15.11.0

(async() => {
  const boxrec = require("boxrec").Boxrec;
  const cookieJar = await boxrec.login('myuser', 'mypass');

  console.log(cookieJar);

  const gennadyGolovkin = await boxrec.getPersonById(cookieJar, 356831);
  console.log(gennadyGolovkin.name); // Gennady Golovkin

})();

Error: 

RequestJar {
  _jar: CookieJar {
    enableLooseMode: true,
    store: { idx: {
      'boxrec.com': {
        '/': {
          PHPSESSID: Cookie="PHPSESSID=oem4oc0fchpqr4157t53slv987; Expires=Thu, 10 Mar 2022 20:01:13 GMT; Max-Age=31536000; Path=/; HttpOnly; hostOnly=true; aAge=3ms; cAge=499ms",
          REMEMBERME: Cookie="REMEMBERME=QXBwXFNlcnZpYdfdfdfzZXJcVXNlcjpjRzkzWlhKclpXNXY6MTY0Njk0MjQ3Mzo1ODNkZmU5N2RjY2IwMWEzMmQ5ZGJjMzVhNTI1YjJkYWY1NmQyODgxODhkMzJlOWFmNGM2YTNjM2IzMjgxMWJi; Expires=Thu, 10 Mar 2022 20:01:13 GMT; Max-Age=31536000; Path=/; HttpOnly; hostOnly=true; aAge=4ms; cAge=500ms"
        }
      }
    } }
  }
}
/var/www/node/node_modules/cheerio/lib/api/traversing.js:30
    return newElems.concat(elem.children.filter(isTag));
                                         ^

TypeError: Cannot read property 'filter' of undefined
    at /var/www/node/node_modules/cheerio/lib/api/traversing.js:30:42
    at Array.reduce (<anonymous>)
    at module.exports.exports.find (/var/www/node/node_modules/cheerio/lib/api/traversing.js:29:30)
    at Function.<anonymous> (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:741:66)
    at step (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:32:23)
    at Object.next (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:13:53)
    at fulfilled (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:4:58)
`

**Comment by mikedidomizio** [comment]
If you could, remove your comment as it has some private cookie stuff and I'm not sure if it'll give access to your account.

You can either resend your comment with it excluded but I also have your comment by email.

**Comment by paulvancity** [comment]
I modified the cookie but will do

**Comment by paulvancity** [comment]
I still seem to be getting the error

├─┬ boxrec@7.1.4
│ ├─┬ boxrec-requests@5.1.1

Node Version: v15.11.0

(async() => {
const boxrec = require("boxrec").Boxrec;
const cookieJar = await boxrec.login('myuser', 'mypass');

console.log(cookieJar);

const gennadyGolovkin = await boxrec.getPersonById(cookieJar, 356831);
console.log(gennadyGolovkin.name); // Gennady Golovkin

})();

Error:

RequestJar {
_jar: CookieJar {
enableLooseMode: true,
store: { idx: {
'boxrec.com': {
'/': {
PHPSESSID: Cookie="PHPSESSID=oem4oc0fchpqr4157t53slv987; Expires=Thu, 10 Mar 2022 20:01:13 GMT; Max-Age=31536000; Path=/; HttpOnly; hostOnly=true; aAge=3ms; cAge=499ms",
REMEMBERME: Cookie="REMEMBERME=REMOVED Expires=Thu, 10 Mar 2022 20:01:13 GMT; Max-Age=31536000; Path=/; HttpOnly; hostOnly=true; aAge=4ms; cAge=500ms"
}
}
} }
}
}
/var/www/node/node_modules/cheerio/lib/api/traversing.js:30
return newElems.concat(elem.children.filter(isTag));
^

TypeError: Cannot read property 'filter' of undefined
at /var/www/node/node_modules/cheerio/lib/api/traversing.js:30:42
at Array.reduce ()
at module.exports.exports.find (/var/www/node/node_modules/cheerio/lib/api/traversing.js:29:30)
at Function. (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:741:66)
at step (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:32:23)
at Object.next (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:13:53)
at fulfilled (/var/www/node/node_modules/boxrec-requests/dist/boxrec-requests.js:4:58)
`

**Comment by mikedidomizio** [comment]
When you get the chance can you bump to 7.1.5.  It should work now but it now comes down to updating what is missing or broken.  For the method you used, it seems to get a lot of the information but the "date of birth" or `born` is now `null`.  Checking BoxRec, it looks like they don't display that information anymore, so that's not broken, but needs to be removed.  Sorry for the long delay.

**Comment by dinomiike** [comment]
Hello! Thank you so much for making this project available to the world! Is Node 8-9 absolutely required? I'm running 17 and when I use the sample code, I get an error saying: `Error: Cookie did not have PHPSESSID and REMEMBERME`. I tried creating a docker image that ran the same script using Node 9 and it gives me a different error.

**Comment by mikedidomizio** [comment]
> Hello! Thank you so much for making this project available to the world! Is Node 8-9 absolutely required? I'm running 17 and when I use the sample code, I get an error saying: `Error: Cookie did not have PHPSESSID and REMEMBERME`. I tried creating a docker image that ran the same script using Node 9 and it gives me a different error.

Hey @dinomiike, the documentation regarding Node versions is probably way out of date.  I'm currently running on 16.13.2 and it's fine.

So that error is thrown [here](https://github.com/boxing/boxrec-requests/blob/master/src/boxrec-requests.ts#L478).  I'm fairly certain I know what the issue is on your side.  It's a bad USERNAME.  The package should have given you a better error message but at some point BoxRec changed and `boxrec-requests` handled the bad username incorrectly.  I will fix that by being generic and not so strict.  (Realistically they shouldn't be differentiating between the two as it allows people to know what accounts exist and do not exist 🤯)

TL;DR I'm fairly certain your username passed as an argument does not exist and/or is undefined

**Comment by dinomiike** [comment]
Thanks! I actually was using the wrong user name! 🤦  I fixed it and updated versions and I'm getting a different a different error now:
```js
/myproject/node_modules/boxrec-requests/dist/boxrec-requests.js:566
                            throw new Error(errorMessage || "unknown error");
                                  ^

Error: unknown error
    at Function.<anonymous> (/myproject/node_modules/boxrec-requests/dist/boxrec-requests.js:566:35)
    at step (/myproject/node_modules/boxrec-requests/dist/boxrec-requests.js:33:23)
    at Object.next (/myproject/node_modules/boxrec-requests/dist/boxrec-requests.js:14:53)
    at fulfilled (/myproject/node_modules/boxrec-requests/dist/boxrec-requests.js:5:58)
```
Seems to indicate the response is a non-200:
```js
boxrec-requests.js:

if (data.statusCode !== 200 || errorMessage !== "") {
  throw new Error(errorMessage || "unknown error");
}
```

My code:
```js
const boxrec = require('boxrec').Boxrec;

(async () => {
  const cookieJar = await boxrec.login(BOXREC_USERNAME, BOXREC_PASSWORD);
  console.log('cookieJar', cookieJar);
  const person = await boxrec.getPersonById(cookieJar, 352);
  console.log('person', person);
})();
```
Although, if I make the request with the same data through [Postman](https://www.postman.com/), it seems to work. Postman indicates an http 200 response from the login endpoint. I must be doing something else wrong. Thanks for responding so quickly!

**Comment by mikedidomizio** [comment]
@dinomiike sorry I fixed one issue and then introduced another 😔.  Should be fixed in `boxrec@7.1.9`

**Comment by dinomiike** [comment]
Thank you! It's working!

**Comment by mikedidomizio** [comment]
Awesome.  I'm closing this issue as it's been broken, then fixed by me, then broken by me and fixed again 😅.

If a new issue arises someone can raise it
